vault-plugins/flant_iam/model/group_test.go

```diff
@@ -1,12 +1,181 @@
 package model
 
 import (
+	"fmt"
 	"testing"
+
+	"github.com/flant/negentropy/vault-plugins/shared/io"
+)
+
+const (
+	groupUUID1 = "00000000-0001-0000-0000-000000000000"
+	groupUUID2 = "00000000-0002-0000-0000-000000000000"
+	groupUUID3 = "00000000-0003-0000-0000-000000000000"
+	groupUUID4 = "00000000-0004-0000-0000-000000000000"
+	groupUUID5 = "00000000-0005-0000-0000-000000000000"
+)
+
+func makeSubjectNotations(subjectType string, uuids []string) []SubjectNotation {
+	validTypes := map[string]struct{}{ServiceAccountType: {}, UserType: {}, GroupType: {}}
+	if _, valid := validTypes[subjectType]; !valid {
+		panic(fmt.Errorf("subject_type %s is invalid", subjectType))
+	}
+	result := make([]SubjectNotation, len(uuids))
+	for i := range uuids {
+		result[i] = SubjectNotation{
+			Type: subjectType,
+			ID:   uuids[i],
+		}
+	}
+	return result
+}
+
+func appendSubjects(subjectsGroups ...[]SubjectNotation) []SubjectNotation {
+	result := []SubjectNotation{}
+	for i := range subjectsGroups {
+		result = append(result, subjectsGroups[i]...)
+	}
+	return result
+}
+
+var (
+	group1 = Group{
+		UUID:            groupUUID1,
+		TenantUUID:      tenantUUID1,
+		Users:           []string{userUUID2, userUUID3},
+		Groups:          []string{groupUUID3},
+		ServiceAccounts: []string{serviceAccountUUID1},
+		Origin:          OriginIAM,
+	}
+	group2 = Group{
+		UUID:       groupUUID2,
+		TenantUUID: tenantUUID1,
+		Users:      []string{userUUID1, userUUID3},
+		Origin:     OriginIAM,
+	}
+	group3 = Group{
+		UUID:            groupUUID3,
+		TenantUUID:      tenantUUID2,
+		Users:           []string{userUUID3, userUUID4},
+		ServiceAccounts: []string{serviceAccountUUID1},
+		Origin:          OriginIAM,
+	}
+	group4 = Group{
+		UUID:            groupUUID4,
+		TenantUUID:      tenantUUID1,
+		Users:           []string{userUUID2, userUUID3},
+		Groups:          []string{groupUUID2, groupUUID3},
+		ServiceAccounts: []string{serviceAccountUUID2, serviceAccountUUID3},
+		Origin:          OriginIAM,
+	}
+	group5 = Group{
+		UUID:       groupUUID5,
+		TenantUUID: tenantUUID1,
+		Groups:     []string{groupUUID2, groupUUID1},
+		Origin:     OriginIAM,
+	}
```

Comment:
Just my 2 cents. I'd say that global static fixtures do not scale well. I'd generate them in the scope of a test or in a special helper for it.

Reply:
this global fixtures are reusable for many tests